Source/WebKit/WebProcess/WebPage/WebPage.cpp

```diff
@@ -5793,11 +5793,13 @@ void WebPage::drawRectToImage(FrameIdentifier frameID, const PrintInfo& printInf
     if (coreFrame) {
 #if PLATFORM(MAC)
         ASSERT(coreFrame->document()->printing() || pdfDocumentForPrintingFrame(coreFrame));
+        RetainPtr<PDFDocument> pdfDocument = pdfDocumentForPrintingFrame(coreFrame);
 #else
         ASSERT(coreFrame->document()->printing());
+        bool pdfDocument = false;
```

Comment:
Please don't use a different type on different branches of the #ifdef. Refactor the code to avoid this.

Reply:
Changed this code to always use a local buffer for print preview.